src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -1214,7 +1214,7 @@ def _k8s_get_credentials_internal(name, acs_info, path, ssh_key_file, overwrite_
 def _handle_merge(existing, addition, key, replace):
     if not addition.get(key, False):
         return
-    if existing[key] is None:
+    if not existing.get(key):
```

Comment:
existing[key] == None / 0 / [] / "" / {} will all trigger this condition, is this the expected behavior ?

Reply:
I'm not an expert on what "valid" vs "invalid" kubeconfigs look like, so I don't know for sure.

I think at the very least a missing key should be treated the same as an empty key (before, a missing key triggered an exception).

I was matching the line above on 1214 which was was doing the same thing (not checking specifically for `== False` or something.

Looking further on in the code, if we don't overwrite `None` / `0` / `""` the code below will blow up I think as we'll try to iterate over it assuming that it's a dict.